src/syntax/ppx_sedlex.ml

```diff
@@ -286,29 +285,49 @@ let codepoint i =
     failwith (Printf.sprintf "Invalid Unicode code point: %i" i);
   i
 
-let regexp_for_char c = Sedlex.chars (Cset.singleton (Char.code c))
-
-let regexp_for_string s =
-  let rec aux n =
-    if n = String.length s then Sedlex.eps
-    else Sedlex.seq (regexp_for_char s.[n]) (aux (succ n))
-  in
-  aux 0
+let char c = Cset.singleton (Char.code c)
+let uchar c = Cset.singleton (Uchar.to_int c)
 
 let err loc s =
   raise (Location.Error (Location.Error.createf ~loc "Sedlex: %s" s))
 
+type encoding = Utf8 | Latin1 | Ascii
+
+let rev_csets_of_string ~loc ~encoding s =
+  match encoding with
+    | Utf8 ->
+        Utf8.fold
+          ~f:(fun acc _ x ->
+            match x with
+              | `Malformed _ -> err loc "Malformed utf-8 string"
+              | `Uchar c -> uchar c :: acc)
+          [] s
+    | Latin1 ->
+        let l = ref [] in
+        for i = 0 to String.length s - 1 do
+          l := char s.[i] :: !l
```

Comment:
According to https://en.wikipedia.org/wiki/ISO/IEC_8859-1, not all values are proper `latin1` characters.

Reply:
```
In 1990, the first version of [Unicode](https://en.wikipedia.org/wiki/Unicode) used the code points of ISO-8859-1 as the first 256 Unicode code points.
```

Reply:
ok!